nixos/modules/services/networking/biboumi.nix

```diff
@@ -0,0 +1,169 @@
+{ config, lib, pkgs, options, ... }:
+with lib;
+let
+  cfg = config.services.biboumi;
+  inherit (config.environment) etc;
+  apparmor = config.security.apparmor.enable;
+  rootDir = "/run/biboumi/mnt-root";
+  stateDir = "/var/lib/biboumi";
+  settingsFile = pkgs.writeText "biboumi.cfg" (
+    generators.toKeyValue {
+      mkKeyValue = k: v:
+        if v == null then ""
+        else generators.mkKeyValueDefault {} "=" k v;
+    } cfg.settings);
+  need_CAP_NET_BIND_SERVICE = cfg.settings.identd_port != 0 && cfg.settings.identd_port < 1024;
+  mkDisableOption = name: mkEnableOption name // { default = true; example = false; };
+in
+{
+  options = {
+    services.biboumi = {
+      enable = mkEnableOption "the Biboumi XMPP gateway to IRC";
+
+      settings = mkOption rec {
+        type = types.attrs;
+        apply = recursiveUpdate default;
+        default = {
+          ca_file = "" + etc."ssl/certs/ca-certificates.crt".source;
+          db_name = "${stateDir}/biboumi.sqlite";
+          identd_port = 113;
+          log_level = 1;
+          policy_directory = "${pkgs.biboumi}/etc/biboumi";
+          port = 5347;
+          realname_customization = true;
+          realname_from_jid = false;
+          xmpp_server_ip = "127.0.0.1";
+        };
+        example = {
+          hostname = "biboumi.example.org";
+        };
+        description = ''
+          See <link xlink:href="https://lab.louiz.org/louiz/biboumi/blob/8.5/doc/biboumi.1.rst">biboumi 8.5</link>
+          for documentation.
+        '';
+      };
+
+      credentialsFile = mkOption {
+        type = types.path;
+        description = ''
+          Path to a configuration file to be merged with the settings.
+          Useful to merge a file which is better kept out of the Nix store
+          because it contains sensible data like <link linkend="opt-services.biboumi.settings">settings.password</link>.
+        '';
+        default = "/dev/null";
+        example = "/run/keys/biboumi.cfg";
+      };
+
+      openFirewall = mkDisableOption "opening of the identd port in the firewall";
```

Comment:
All `openFirewall` options should be false by default such that the user consciously has to decide to open it, which might open their server to security problems. The only exception to this is `services.openssh.openFirewall`, whose default we can't change to `false` because it would lock so many people out of their systems, but ideally we would change this.

Reply:
@Infinisil I've changed it.